File: src/main/java/com/devmpv/config/RepositoryConfig.java
package com.devmpv.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurerAdapter;

import com.devmpv.model.Board;
import com.devmpv.model.Message;
import com.devmpv.model.Thread;

/**
 * REST repositories configuration
 * 
 * @author devmpv
 */
@Configuration
public class RepositoryConfig extends RepositoryRestConfigurerAdapter {
    @Override
    public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config) {
	config.exposeIdsFor(Board.class, Thread.class, Message.class);
    }
}

File: src/main/kotlin/com/devmpv/config/RepositoryConfig.kt
package com.devmpv.config

import org.springframework.context.annotation.Configuration
import com.devmpv.model.Board
import com.devmpv.model.Message
import com.devmpv.model.Thread
import org.springframework.context.annotation.Bean
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer

/**
 * REST repositories configuration
 *
 * @author devmpv
 */
@Configuration
class RepositoryConfig {

    @Bean
    fun repositoryRestConfigurer(): RepositoryRestConfigurer =
        RepositoryRestConfigurer.withConfig { it ->
            it.exposeIdsFor(Board::class.java, Thread::class.java, Message::class.java)
        }
}
